src/main/java/com/google/sps/servlets/DataFromDatabase.java

```diff
@@ -0,0 +1,134 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/data-from-db")
+public class DataFromDatabase extends HttpServlet {
+
+  DatabaseClient dbClient;
+  private String[] selectedTables;
+
+  public void init() {
+    Spanner spanner = SpannerOptions.newBuilder().build().getService();
+    DatabaseId db = DatabaseId.of("play-user-data-beetle", "test-instance", "example-db");
+    this.dbClient = spanner.getDatabaseClient(db);
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    for (String table : selectedTables) {
+      String colQuery = "SELECT column_name, spanner_type, is_nullable FROM information_schema.columns WHERE table_name = '" + table + "'";
+
+      try (ResultSet resultSet =
+        dbClient.singleUse().executeQuery(Statement.of(colQuery))) {
+
+        List<String> colnames = new ArrayList<>();
+        List<String> spannerTypes = new ArrayList<>();
+        List<String> nullables = new ArrayList<>();
```

Comment:
Would it be clearer if we had an object representation of "ColumnMetadata"? If so, take a look at AutoValue it may save you some time: https://github.com/google/auto/blob/master/value/userguide/index.md

Reply:
I think so! I looked into this today and will work on implementing it tomorrow.

Reply:
Doing this in separate PR

Reply:
Done.